packages/dds/tree/src/core/rebase/utils.ts

```diff
@@ -620,3 +621,34 @@ export function findCommonAncestor<T extends { parent?: T }>(
 	}
 	return undefined;
 }
+
+export function replaceChange<TChange>(
+	commit: GraphCommit<TChange>,
+	change: TChange,
+): GraphCommit<TChange> {
+	const output = { ...commit, change };
+	Rollback.set(output, undefined);
+	return output;
+}
+
+/** Associates rollback data with commits */
+namespace Rollback {
+	const map = new WeakMap<GraphCommit<unknown>, TaggedChange<unknown, RevisionTag>>();
+
+	export function get<TChange>(
+		commit: GraphCommit<TChange>,
+	): TaggedChange<TChange, RevisionTag> | undefined {
+		return map.get(commit) as TaggedChange<TChange, RevisionTag> | undefined;
+	}
+
+	export function set<TChange>(
+		commit: GraphCommit<TChange>,
+		rollback: TaggedChange<TChange, RevisionTag> | undefined,
+	): void {
+		if (rollback === undefined) {
+			map.delete(commit);
```

Comment:
why do this rather than just have a separate delete function? it seems like replaceChange explicitly deletes and rollbackFromCommit only ever passes in a rollback

Reply:
TBH, mostly because `delete` is a reserved keyword in JS that you can't use a function name, so I just rolled it into `set` and then it's less boilerplate too (two functions instead of three). We could have a delete function and call it `remove`, or something, if you think that's better.

Reply:
The only reason we need these functions in the first place btw is because they enforce that the output TChange is the same as the input TChange when doing a get (we know this, because it was the case when we did the set). If it weren't for that, we could just get rid of all these and just use the map directly. I mean, we could do that anyway, and just cast at the read sites, but that feels slightly worse.